From 0561a23f1adf524729138480f2a8a836fd5145a7 Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Thu, 3 Sep 2015 00:02:22 +0800
Subject: [PATCH 1295/1594] MLK-11491 ARM: imx: dis-connect vddhigh_in and
 vddsnvs_in for imx6ul

commit d9dd229076c09173b674becda8b57b94c61089bb from
git://git.freescale.com/imx/linux-2.6-imx.git

To achieve lower power number in low power idle, on the boards
which has higher vddhigh_in than vddsnvs_in, need to dis-connect
vddhigh_in and vddsnvs_in internally to make vddhigh_in's power
low enough to meet the design target.

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 arch/arm/mach-imx/imx6ul_low_power_idle.S |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/arch/arm/mach-imx/imx6ul_low_power_idle.S b/arch/arm/mach-imx/imx6ul_low_power_idle.S
index 5db724e..26bb83d 100644
--- a/arch/arm/mach-imx/imx6ul_low_power_idle.S
+++ b/arch/arm/mach-imx/imx6ul_low_power_idle.S
@@ -310,6 +310,11 @@
 	orr	r7, r7, #0x6000
 	str	r7, [r10, #0x150]
 
+	/* disconnect vdd_high_in and vdd_snvs_in */
+	ldr	r7, [r10, #0x150]
+	orr	r7, r7, #0x1000
+	str	r7, [r10, #0x150]
+
 	.endm
 
 	.macro anatop_exit_idle
-- 
1.7.5.4

